### PR TITLE
cleanup: unused kubelet serving certificate rotation code

### PR DESCRIPTION
**Pull request by cameronmeissner**
**What type of PR is this?**

**What this PR does / why we need it**:
removes unused code for kubelet serving certificate rotation

verified all scenarios are working properly via manual e2e validation:

enabled:
```
2024-10-24T19:08:43.8503328+00:00: AKS.WindowsCSE.ConfigureKubeletServingCertificateRotation - EnableKubeletServingCertificateRotation: True. Configure kubelet config args and node labels for serving certificate rotation
2024-10-24T19:08:44.0534672+00:00: Kubelet serving certificate rotation is enabled, will add node label if needed
2024-10-24T19:08:44.0534672+00:00: adding label kubernetes.azure.com/kubelet-serving-ca=cluster to kubelet node labels...
```

disabled by nodepool tag:
```
2024-10-24T20:22:38.9937516+00:00: Copying provisioning scripts
2024-10-24T20:22:39.0093771+00:00: Moving various log collect scripts and depencencies
2024-10-24T20:22:39.0093771+00:00: AKS.WindowsCSE.ConfigureKubeletServingCertificateRotation - EnableKubeletServingCertificateRotation: True. Configure kubelet config args and node labels for serving certificate rotation
2024-10-24T20:22:39.1031908+00:00: Kubelet serving certificate rotation is disabled by nodepool tags, will reconfigure kubelet flags and node labels
```

disabled:
```
2024-10-24T20:49:18.5836931+00:00: Moving various log collect scripts and depencencies
2024-10-24T20:49:18.5993221+00:00: AKS.WindowsCSE.ConfigureKubeletServingCertificateRotation - EnableKubeletServingCertificateRotation: False. Configure kubelet config args and node labels for serving certificate rotation
2024-10-24T20:49:18.5993221+00:00: Kubelet serving certificate rotation is disabled, nothing to configure
2024-10-24T20:49:18.5993221+00:00: AKS.WindowsCSE.WriteKubeClusterConfig - Start to write KubeCluster Config. WindowsPauseImageURL: mcr.microsoft.com/oss/kubernetes/pause:3.9-hotfix-20230808
```

disabled also with nodepool tag:
```
2024-10-24T21:01:19.0155586+00:00: Moving various log collect scripts and depencencies
2024-10-24T21:01:19.0311863+00:00: AKS.WindowsCSE.ConfigureKubeletServingCertificateRotation - EnableKubeletServingCertificateRotation: False. Configure kubelet config args and node labels for serving certificate rotation
2024-10-24T21:01:19.0311863+00:00: Kubelet serving certificate rotation is disabled, nothing to configure
2024-10-24T21:01:19.0311863+00:00: AKS.WindowsCSE.WriteKubeClusterConfig - Start to write KubeCluster Config. WindowsPauseImageURL: mcr.microsoft.com/oss/kubernetes/pause:3.9-hotfix-20230808
```

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: parts/windows/kuberneteswindowssetup.ps1 [hunk 1]
```diff
@@ -296,10 +296,7 @@ try
 
     # NOTE: this function MUST be called before Write-KubeClusterConfig since it has the potential
     # to mutate both kubelet config args and kubelet node labels.
-    Disable-KubeletServingCertificateRotationForTags
-
-    # TODO(cameissner): uncomment once CSE package has been updated
-    # Configure-KubeletServingCertificateRotation
+    Configure-KubeletServingCertificateRotation
```
Comment (UtheMan) [inline parts/windows/kuberneteswindowssetup.ps1:299]: The comment above - does it apply to both functions (i mean the one you removed and tthis new uncommented one)?

**Review by UtheMan — APPROVED** [review]

**Review by r2k1 — APPROVED** [review]

**Comment by cameronmeissner** [comment]
will not merge until a new script package has been released containing this fix: https://github.com/Azure/AgentBaker/pull/5143

File: parts/windows/kuberneteswindowssetup.ps1 (hunk 1 above)
Comment (cameronmeissner) [reply parts/windows/kuberneteswindowssetup.ps1:299]: yep, that's why the comment's staying in

**Comment by cameronmeissner** [comment]
waiting on e2e verification, script package was bumped containing the fix from a separate PR

**Pull request merged**